client/pages/results.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import queryString from 'query-string';
+import Nav from '../components/navbar';
+import Card from 'react-bootstrap/Card';
+
+export default class Results extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({
+      address: queryString.parse(this.props.location.search).address,
+      input: '',
+      walletData: {
+        chain_stats: {
+          tx_count: 0,
+          funded_txo_sum: 0,
+          spent_txo_sum: 0
+        }
+      },
+      transactionData: [
+        {
+          txid: '',
+          status: {
+            block_height: 0
+          }
+        },
+        {
+          txid: '',
+          status: {
+            block_height: 0
+          }
+        },
+        {
+          txid: '',
+          status: {
+            block_height: 0
+          }
+        }
+      ]
+    });
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.unlisten = this.props.history.listen((location, action) => {
+      this.setState({ address: queryString.parse(location.search).address });
+      this.fetchData(queryString.parse(location.search).address);
+    });
+    this.fetchData(this.state.address);
+  }
```

Comment:
this is fine, just be sure to add another lifecycle method to cancel the listener when the component unmounts

```js
componentWillUnmount() {
  this.unlisten();
}
```